tests/test_odin_sequencer.py

```diff
@@ -4,26 +4,20 @@
 
 import pytest
 
+from pathlib import Path
 from odin_sequencer import CommandSequenceManager, CommandSequenceError
 
 @pytest.fixture
-def make_seq_manager(shared_datadir):
+def create_paths(shared_datadir):
```

Comment:
Why is the creation of a CommandSequenceManager delegated to each test case rather than being done inside the factory function returned by the fixture?

Reply:
The constructor of `CommandSequenceManager` no loner has a `file_or_files` parameter to allow for file modules to be optionally passed and loaded at the time of instantiating a sequence manager object. An object must be first instantiated and then its `load_module` function has to be called in order for modules to be loaded. The factory function was creating a list of string paths if file names were provided to it by the test functions and was passing this to the `CommandSequenceManager` constructor to instantiate an object. However, because the constructor is no longer taking file modules, the factory function is only going to be instantiating sequencer manager objects and thought that the same can be achieved directly in the test functions with one line of code rather than having to call a method.

Not sure if there are any associated issues to doing this, but I am happy to change it if needed.

Reply:
Ah OK I'd missed that change. I would suggest reinstating the ability to pass a `file_or_files` (or equivalent `paths` as per `load_module()`. This makes it simpler to instantiate a trivial `CommandSequenceManager`, which is what I had intended originally. Is there a technical reason why this is no longer possible? It should be feasible to do either (or both) I think? 

I wonder whether the name `load_module()` is no-longer as descriptive as it could be?

Reply:
Looking at it now, I think that it should be feasible to accomplish both. Originally the `load_module` function was only designed to take one path at a time, so callers (including the constructor) had to loop over a list of files, and provide one path at a time to it (like below) if multiple modules had to be loaded. 
```Python
for file in path_or_paths:
    self.load_module(file, resolve=False)

self.resolve()
```
I thought that it might be better to have the `load_module` function take multiple paths and handle the looping inside it, allowing callers to load multiple modules without having to iterate over a list and call the `load_module` function multiple times. After refactoring the `load_module` function to do the looping, I felt that part of the logic in the constructor is duplicate (as it is also doing the same looping), so I decided to get rid of the ability to pass files/ paths to the constructor. Thinking about it now, I can see that I was wrong because the looping isn't needed anymore as multiple paths can be passed now..

Before I do any changes, I want to double check with you whether having the `load_module` function take multiple paths is a good idea or not? If I change the `load_module` function to how it was, callers will have to loop through paths and call it multiple times if multiple modules need to be loaded.

Reply:
I think having `load_module` work in the way you have now implemented (with looping) is absolutely fine and the correct thing to do. I think we simply add add an argument to the constructor, which defaults to `None` and, if set, calls `load_module` in the same way. Then you can choose to either pass in a path or list of paths to the constructor, or initialise it empty and call `load_module`, or do both 😜 .

I completely agree that we should avoid forcing the user to loop when passing a path or paths in.

I think it would be worth changing the name of `load_module` (maybe to just `load`?? any other ideas??) since its behaviour has changed and it is becoming a more public interface to the manager.

Reply:
Thanks for you sharing your opinion on this! I will refactor it and have a think about names ideas for the `load_module` function.